protocols/kad/src/behaviour.rs

```diff
@@ -2036,6 +2067,10 @@ where
         }
     }
 
+    fn on_new_kademlia_node(&mut self) {
+        let _ = self.bootstrap();
```

Comment:
We should store the current query ID here and avoid kicking off queries in parallel. Otherwise, I think this will be recursive (we discover new nodes as part of bootstrapping).

Reply:
I stored the query ID in the bootstrap function and made it completed in this function. 
If it should be stored here, then what is the place to set it to completed?